File: lambda_deploy.js
// https://aws.amazon.com/blogs/compute/new-deployment-options-for-aws-lambda/
console.log('Loading function');
var AWS = require('aws-sdk');
var lambda = new AWS.Lambda();
exports.handler = function(event, context) {
    key = event.Records[0].s3.object.key
    bucket = event.Records[0].s3.bucket.name
    version = event.Records[0].s3.object.versionId
    if (bucket == "ask-ava" && key == "ask-ava.zip" && version) {
        var functionName = "ask-ava";
        console.log("uploaded to lambda function: " + functionName);
        var params = {
            FunctionName: functionName,
            S3Key: key,
            S3Bucket: bucket,
            S3ObjectVersion: version
        };
        lambda.updateFunctionCode(params, function(err, data) {
            if (err) {
                console.log(err, err.stack);
                context.fail(err);
            } else {
                console.log(data);
                context.succeed(data);
            }
        });
    } else {
        context.succeed("skipping zip " + key + " in bucket " + bucket + " with version " + version);
    }
};

File: lambda_deploy.py
"""
lambda deploy
"""

from __future__ import print_function
import boto3

WATCH = [{'bucket'  : 'ask-ava',
          'key'     : 'ask-ava.zip',
          'function': 'ask-ava'
         }]

def lambda_handler(event, context):
    """ lambda handler """
    status = None
    key = event['Records'][0]['s3']['object']['key']
    bucket = event['Records'][0]['s3']['bucket']['name']

    client = boto3.client('lambda')

    for item in WATCH:
        if bucket in item['bucket'] and key in item['key']:
            response = client.update_function_code(
                FunctionName=item['function'],
                S3Bucket=item['bucket'],
                S3Key=item['key']
            )
            status = True
    return status
